rosbag2_py/src/rosbag2_py/format_service_info.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 Sony Group Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sstream>
+
+#include "format_service_info.hpp"
+
+std::string
+format_service_info(
+  std::vector<std::shared_ptr<rosbag2_cpp::rosbag2_service_info_t>> & service_info_list)
+{
+  std::stringstream info_stream;
+  int indentation_spaces = 21;
```

Comment:
Comments from MichaelOrlov

Out of curiosity why indentation is 21?
For format_bag_meta_data indentation is 19.

Reply:
@MichaelOrlov 

```
Topic information: Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 0 | Serialization Format: cdr
                   Topic: /parameter_events | Type: rcl_interfaces/msg/Log | Count: 0 | Serialization Format: cdr
Service information: Service: /add_two_ints | Type: example_interfaces/srv/AddTwoInts | Event Count: 30 | Serialization Format: cdr
                     Service: ...
```

This length depends on the length of the prefix string.  
One is `Topic information: `.
Another is `Service information: `

Reply:
@Barry-Xu-2018 Could you please derive `indentation_spaces` from those string?
For instance:
```cpp
  constexpr const char service_information_str[] = "Service information: ";
  info_stream << service_information_str;
  int indentation_spaces = sizeof(service_information_str);
```
It would be clear and self explained for others.

Reply:
Thanks. I will modify code like your suggestion.